smdebug/tensorflow/base_hook.py

```diff
@@ -180,7 +180,8 @@ def _get_default_collections(self):
         return DEFAULT_TF_COLLECTIONS
 
     def export_collections(self):
-        assert self._prepared_tensors[self.mode]
+        if not self.tape:
```

Comment:
You'll need to define the `tape` attribute in the base class, rather than KerasHook, if methods here are referring to it. This will cause an AssertionError if the method is called on a SessionHook where `tape` is not defined.